src/odemis/gui/cont/correlation.py

```diff
@@ -0,0 +1,280 @@
+
+# -*- coding: utf-8 -*-
+
+"""
+@author Patrick Cleeve
+
+Copyright © 2023, Delmic
+
+Handles the controls for correlating two (or more) streams together.
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License version 2 as published by the Free
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+import logging
+import numpy
+import wx
+# IMPORTANT: wx.html needs to be imported for the HTMLWindow defined in the XRC
+# file to be correctly identified. See: http://trac.wxwidgets.org/ticket/3626
+# This is not related to any particular wxPython version and is most likely permanent.
+import wx.html
+import odemis.acq.stream as acqstream
+import odemis.gui.model as guimod
+from odemis import model
+from odemis.acq.stream import StaticStream
+from odemis.gui.util import call_in_wx_main
+
+# TODO: move to more approprate location
+def update_image_in_views(s: StaticStream, views: list) -> None:
+    """Force update the static stream in the selected views
+    param s: (StaticStream) the static stream to update
+    param views: (list[View]) the list of views to update"""
+    v: guimod.View 
+    for v in views:
+        for sp in v.stream_tree.getProjections():  # stream or projection
+            if isinstance(sp, acqstream.DataProjection):
+                st = sp.stream
+            else:
+                st = sp
+            if st is s:
+                sp._shouldUpdateImage()
+
+
+# TODO: add toggle for enablling moving the streams. 
+# If not enabled, the streams dont move, use standard controls for mouse / keyboard events...
+class CorrelationController(object):
+
+    def __init__(self, tab_data, panel, tab, viewports):
+        """
+        param tab_data: (MicroscopyGUIData) the representation of the microscope GUI
+        param panel (wx._windows.Panel): the panel containing the UI controls
+        param tab: (Tab): the tab which should show the data
+        param viewports: (ViewPorts): the tab view ports
+        """
+        
+        self._tab_data_model = tab_data
+        self._main_data_model = tab_data.main
+        self._panel = panel
+        self._tab = tab
+        self._viewports = viewports
+
+        # selected stream
+        self.selectedStream = model.VigilantAttribute(None)
```

Comment:
That VA should be an attribute of `CryoCorrelationGUIData` (so it's on `tab_data`).

Reply:
Is this what you mean? I think it should stay here, as it is only used inside this widget. It makes it transferable to other tabs in the future.

Reply:
OK, understand. I've moved it into GUIData